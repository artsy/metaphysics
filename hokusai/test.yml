version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
