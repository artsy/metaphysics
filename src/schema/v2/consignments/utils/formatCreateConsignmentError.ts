import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
