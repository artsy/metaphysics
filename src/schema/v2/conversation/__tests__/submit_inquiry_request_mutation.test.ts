import gql from "lib/gql"
import { runAuthenticatedQuery } from "schema/v2/test/utils"

describe("SubmitInquiryRequestMutation", () => {
  const userByIDLoader = jest.fn(() => {
    return Promise.resolve({
      id: "rob-ross",
      name: "Rob Ross",
      email: "rob@ross.com",
    })
  })

  const artistLoader = jest.fn(() => {
    return Promise.resolve({
      id: "bob-ross",
      name: "Bob Ross",
      birthday: "1929",
    })
  })

  describe("when question is present", () => {
    it("calls gravity loader with questions and address", async () => {
      const mutation = gql`
        mutation {
          submitInquiryRequestMutation(
            input: {
              inquireableID: "artwork-id"
              inquireableType: "Artwork"
              questions: [
                { questionID: "price_and_availability" }
                {
                  questionID: "shipping_quote"
                  details: "{ \\"address\\": \\"112 Main st Brooklyn NY 11211\\"}"
                }
              ]
            }
          ) {
            inquiryRequest {
              inquireable {
                __typename
                ... on Artwork {
                  title
                  artists {
                    name
                    birthday
                  }
                }
              }
              inquirer {
                name
                email
              }
              shippingLocation {
                address
                city
                country
              }
              questions
            }
          }
        }
      `

      const submitArtworkInquiryRequestLoader = jest.fn(() => {
        return Promise.resolve({
          inquireable_type: "Artwork",
          inquireable: {
            id: "bob-ross",
            title: "Happy little accident",
            artists: [{ id: "bob-ross" }],
          },
          inquirer: {
            id: "rob-ross",
          },
          inquiry_questions: ["price_and_availability", "shipping_quote"],
          inquiry_shipping_location: {
            address: "112 Main st Brooklyn NY 11211",
          },
        })
      })

      const context = {
        submitArtworkInquiryRequestLoader,
        userByIDLoader,
        artistLoader,
      }

      expect.assertions(4)
      const { submitInquiryRequestMutation } = await runAuthenticatedQuery(
        mutation,
        context
      )
      expect(submitArtworkInquiryRequestLoader).toHaveBeenCalledWith({
        artwork: "artwork-id",
        inquiry_questions: ["price_and_availability", "shipping_quote"],
        inquiry_shipping_location: {
          address: "112 Main st Brooklyn NY 11211",
        },
      })
      expect(userByIDLoader).toHaveBeenCalledWith("rob-ross")
      expect(artistLoader).toHaveBeenCalledWith("bob-ross")
      expect(submitInquiryRequestMutation).toMatchSnapshot()
    })
  })
  describe("when only message is present", () => {
    it("calls gravity loader with only message", async () => {
      const mutation = gql`
        mutation {
          submitInquiryRequestMutation(
            input: {
              inquireableID: "artwork-id"
              inquireableType: "Artwork"
              message: "do you have sunset paintings?"
            }
          ) {
            inquiryRequest {
              inquireable {
                __typename
                ... on Artwork {
                  title
                  artists {
                    name
                    birthday
                  }
                }
              }
              inquirer {
                name
                email
              }
              shippingLocation {
                address
                city
                country
              }
              questions
            }
          }
        }
      `

      const submitArtworkInquiryRequestLoader = jest.fn(() => {
        return Promise.resolve({
          inquireable_type: "Artwork",
          inquireable: {
            id: "bob-ross",
            title: "Happy little accident",
            artists: [{ id: "bob-ross" }],
          },
          inquirer: {
            id: "rob-ross",
          },
        })
      })

      const context = {
        submitArtworkInquiryRequestLoader,
        userByIDLoader,
        artistLoader,
      }

      expect.assertions(4)
      const { submitInquiryRequestMutation } = await runAuthenticatedQuery(
        mutation,
        context
      )
      expect(submitArtworkInquiryRequestLoader).toHaveBeenCalledWith({
        artwork: "artwork-id",
        message: "do you have sunset paintings?",
      })
      expect(userByIDLoader).toHaveBeenCalledWith("rob-ross")
      expect(artistLoader).toHaveBeenCalledWith("bob-ross")
      expect(submitInquiryRequestMutation).toMatchSnapshot()
    })
  })
})
