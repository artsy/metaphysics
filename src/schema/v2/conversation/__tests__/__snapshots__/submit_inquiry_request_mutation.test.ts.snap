// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitInquiryRequestMutation when only message is present calls gravity loader with only message 1`] = `
{
  "inquiryRequest": {
    "inquireable": {
      "__typename": "Artwork",
      "artists": [
        {
          "birthday": "1929",
          "name": "Bob Ross",
        },
      ],
      "title": "Happy little accident",
    },
    "inquirer": {
      "email": "rob@ross.com",
      "name": "Rob Ross",
    },
    "questions": null,
    "shippingLocation": null,
  },
}
`;

exports[`SubmitInquiryRequestMutation when question is present calls gravity loader with questions and address 1`] = `
{
  "inquiryRequest": {
    "inquireable": {
      "__typename": "Artwork",
      "artists": [
        {
          "birthday": "1929",
          "name": "Bob Ross",
        },
      ],
      "title": "Happy little accident",
    },
    "inquirer": {
      "email": "rob@ross.com",
      "name": "Rob Ross",
    },
    "questions": [
      "price_and_availability",
      "shipping_quote",
    ],
    "shippingLocation": {
      "address": "112 Main st Brooklyn NY 11211",
      "city": null,
      "country": null,
    },
  },
}
`;
